### PR TITLE
Add side-by-side rendering as global setting

### PR DESCRIPTION
**Pull request by jess-x**
## References
https://github.com/jupyterlab/jupyterlab/issues/11376

## Code changes

- Added `sideBySideRendering` to notebook setting schema
- Made `renderSideBySide` in `notebook-extension` listen to `sideBySideRendering` to notebook setting

## User-facing changes
- Enabled setting side-by-side rendering as default 

## Backwards-incompatible changes
I am not aware of any.

### CODE REVIEW

**Comment by echarles** [comment]
The `Linux JS Flaky Tests / JS (js-services)` has failed. I have relaunched the CI but still failing with same errors. This is also the case for other PR (eg https://github.com/jupyterlab/jupyterlab/runs/4304107523?check_suite_focus=true), so we can consider as non related to the changes of this PR which don't touch the services

```
test/kernel/ikernel.spec.ts
  Kernel.IKernel
    #restart()
      ✕ should restart and resolve with a valid server response
test/session/session.spec.ts
  session
    Session.listRunning()
      ✕ should yield a list of valid session models
```

**Review by echarles — APPROVED** [review]
Tested and LGTM. Thx @jess-x

File: packages/notebook-extension/src/index.ts [hunk 1]
```diff
@@ -2319,9 +2323,8 @@ function addCommands(
         if (wideget) {
           if (Private.renderSideBySide) {
             return NotebookActions.renderSideBySide(wideget.content);
-          } else {
-            return NotebookActions.renderNotSideBySide(wideget.content);
           }
+          return NotebookActions.renderNotSideBySide(wideget.content);
```
Comment (jtpio) [inline packages/notebook-extension/src/index.ts:2327]: Mind taking the opportunity to fix the small typo?

```suggestion
          return NotebookActions.renderNotSideBySide(widget.content);
```

Thanks!

Comment (jess-x) [reply packages/notebook-extension/src/index.ts]: 😀 Typo fixed

Comment (jtpio) [reply packages/notebook-extension/src/index.ts]: Thanks!

**Comment by echarles** [comment]
Merged. Thx @jess-x for the PR and @jtpio for the review.

**Comment by jess-x** [comment]
@echarles thanks so much for merging this PR. I took the liberty to label this as `enhancement`, but please correct me if this should be part of the next patch. I could backport this to 3.2.x if that's the case

**Comment by echarles** [comment]
@meeseeksdev please backport to 3.2.x

**Pull request merged**